src/main/java/com/app/controllers/LeaderboardController.java

```diff
@@ -87,10 +107,12 @@ public void displayTop5Results() {
               COMPETITION_CONTROLLER.distanceToArray(style, GenderType.OTHER).length);
       DistanceType distance = DistanceType.values()[distanceInput - 1];
 
-      ArrayList<ResultModel> top5 = new ArrayList<>();
-      Collections.addAll(top5, findTop(style, distance, 5));
-
-      VIEW.displayTopResults(arrayWithResultToDisplay(top5));
+      try {
+        ArrayList<ResultModel> top5 = findTop(style, distance, 5);
+        VIEW.displayTopResults(arrayWithResultToDisplay(top5));
+      } catch (NullPointerException e) {
```

Comment:
we can consider a different exception here - one with a more business orientated name